src/NServiceBus.Core/Pipeline/Incoming/InvokeHandlerContext.cs

```diff
@@ -51,6 +52,11 @@ public InvokeHandlerContext(MessageHandler handler, string messageId, string rep
         public SynchronizedStorageSession SynchronizedStorageSession => Get<SynchronizedStorageSession>();
 
         /// <summary>
+        /// Token
+        /// </summary>
+        public CancellationToken CancellationToken => this.Get<CancellationTokenSource>().Token;
```

Comment:
Should we cache the token ?, so in other words call `this.Get<CancellationTokenSource>().Token` only once.